File: flight_software/modules/devices/arduino.py


class Arduino(Device):

    def __init__(self, name: str, addr: hex):
        super(name)
        self.addr = addr
    
    def status(self) -> bool:
        pass

    def reset(self) -> bool:
        pass

    """
    Read data from the Arduino
    """
    def read(self) -> bytes:
        pass

    """
    Write data to the Arduino
    """
    def write(self, msg: bytes) -> bytes:
        pass
    

File: src/modules/drivers/Arduino.cpp
#include <Logger/logger_util.h>
#include <flight/modules/drivers/Arduino.hpp>
#include <flight/modules/drivers/PseudoSensor.hpp>
#include <flight/modules/drivers/PseudoValve.hpp>

void Arduino::reset(){
    if (name == "PseudoSensor"){
        arduino = new PseudoSensor();
    } else if(name == "PseudoValve") {
        arduino = new PseudoValve();
    } else {
        log("REEEE UNKNOWN ARDUINO TYPE");
    }
}

unsigned char* Arduino::read() {
    return arduino->read();
}

void Arduino::write(unsigned char* msg) {
    arduino->write(msg);
}
